prov/tcp/src/tcpx_progress.c

```diff
@@ -42,6 +42,42 @@
 #include <ofi_util.h>
 #include <ofi_iov.h>
 
+int tcpx_ep_shutdown_report(struct tcpx_ep *ep, fid_t fid, int err)
+{
+	struct fi_eq_cm_entry cm_entry = {0};
+	struct fi_eq_err_entry err_entry = {0};
+	int ret = FI_SUCCESS;
+
+	fastlock_acquire(&ep->cm_state_lock);
+	if (ep->cm_state == TCPX_EP_CONN_SHUTDOWN)
+		goto success;
+
+	if (err) {
+		err_entry.fid = fid;
+		err_entry.context = fid->context;
+		err_entry.err = -err;
+		if (fi_eq_write(&ep->util_ep.eq->eq_fid, FI_SHUTDOWN,
+				&err_entry, sizeof(err_entry),
+				UTIL_FLAG_ERROR) < 0) {
+			ret = (int) -errno;
+			goto err;
+		}
+		goto success;
```

Comment:
restructure to remove goto.  Place code after if (err) check above into else clause.